image-rauc.c

```diff
@@ -102,8 +103,10 @@ static int rauc_parse(struct image *image, cfg_t *cfg)
 		free(part);
 		return -EINVAL;
 	}
-	part->partition_type = RAUC_KEY;
-	list_add_tail(&part->list, &image->partitions);
+	if (strncmp(pkcs11_prefix, part->image, strlen(pkcs11_prefix))) {
+		part->partition_type = RAUC_KEY;
+		list_add_tail(&part->list, &image->partitions);
+	}
```

Comment:
The part object is not needed for pkcs11 and is leaked. Add the check before it is created and skip creating it.